src/pkg/systemartifact/dao/dao.go

```
@@ -0,0 +1,134 @@
+
+	// List all the system artifact records that match the criteria specified
+	// within the query.
+	List(ctx context.Context, query *q.Query) ([]*model.SystemArtifact, error)
+
+	// Size returns the sum of all the system artifacts.
+	Size(ctx context.Context) (int64, error)
+}
+
+// NewSystemArtifactDao returns an instance of the system artifact dao layer
+func NewSystemArtifactDao() DAO {
+	return &systemArtifactDAO{}
+}
+
+// The default implementation of the system artifact DAO.
+type systemArtifactDAO struct{}
+
+func (*systemArtifactDAO) Create(ctx context.Context, artifactRecord *model.SystemArtifact) (int64, error) {
+	ormer, err := orm.FromContext(ctx)
+	if err != nil {
+		return 0, err
+	}
+	id, err := ormer.Insert(artifactRecord)
+	if err != nil {
+		if e := orm.AsConflictError(err, "system artifact with repository name %s and digest %s already exists",
+			artifactRecord.Repository, artifactRecord.Digest); e != nil {
+			err = e
+		}
+		return int64(0), err
+	}
+	return id, nil
+}
+
+func (*systemArtifactDAO) Get(ctx context.Context, vendor string, repository string, digest string) (*model.SystemArtifact, error) {
+	ormer, err := orm.FromContext(ctx)
+
+	if err != nil {
+		return nil, err
+	}
+
+	sa := model.SystemArtifact{Repository: repository, Digest: digest, Vendor: vendor}
+
+	err = ormer.Read(&sa, "vendor", "repository", "digest")
+
+	if err != nil {
+		if e := orm.AsNotFoundError(err, "system artifact with repository name %s and digest %s not found",
+			repository, digest); e != nil {
+			err = e
+		}
+		return nil, err
+	}
+
+	return &sa, nil
+}
+
+func (*systemArtifactDAO) Delete(ctx context.Context, vendor string, repository string, digest string) error {
+	ormer, err := orm.FromContext(ctx)
+
+	if err != nil {
+		return err
+	}
+
+	sa := model.SystemArtifact{
+		Repository: repository,
+		Digest:     digest,
+		Vendor:     vendor,
+	}
+
+	_, err = ormer.Delete(&sa, "vendor", "repository", "digest")
+
+	return nil
```

why not return the err?